pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -32,114 +32,185 @@ func NewConfigurableInterpreter(informer genericmanager.SingleClusterInformerMan
 		return
 	}
-	revised, err = c.luaVM.ReviseReplica(object, replica, luaScript)
+
+	script := accessor.GetReplicaRevisionLuaScript()
+	if len(script) == 0 {
+		enabled = false
+		return
+	}
+
+	revised, err = c.luaVM.ReviseReplica(object, replica, script)
 	return
 }
 
 // Retain returns the objects that based on the "desired" object but with values retained from the "observed" object.
 func (c *ConfigurableInterpreter) Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, enabled bool, err error) {
 	klog.V(4).Infof("Retain object: %v %s/%s with configurable interpreter.", desired.GroupVersionKind(), desired.GetNamespace(), desired.GetName())
-	luaScript, enabled := c.getInterpreter(desired.GroupVersionKind(), configv1alpha1.InterpreterOperationRetain)
+
+	accessor, enabled := c.getCustomAccessor(desired.GroupVersionKind())
 	if !enabled {
 		return
 	}
-	retained, err = c.luaVM.Retain(desired, observed, luaScript)
+
+	script := accessor.GetRetentionLuaScript()
+	if len(script) == 0 {
+		enabled = false
+		return
+	}
+
+	retained, err = c.luaVM.Retain(desired, observed, script)
 	return
 }
 
 // AggregateStatus returns the objects that based on the 'object' but with status aggregated.
 func (c *ConfigurableInterpreter) AggregateStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (status *unstructured.Unstructured, enabled bool, err error) {
 	klog.V(4).Infof("Aggregate status of object: %v %s/%s with configurable interpreter.", object.GroupVersionKind(), object.GetNamespace(), object.GetName())
-	luaScript, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationAggregateStatus)
+	accessor, enabled := c.getCustomAccessor(object.GroupVersionKind())
 	if !enabled {
 		return
 	}
-	status, err = c.luaVM.AggregateStatus(object, aggregatedStatusItems, luaScript)
+
+	script := accessor.GetStatusAggregationLuaScript()
+	if len(script) == 0 {
+		enabled = false
+		return
+	}
+
+	status, err = c.luaVM.AggregateStatus(object, aggregatedStatusItems, script)
 	return
 }
 
 // GetDependencies returns the dependent resources of the given object.
 func (c *ConfigurableInterpreter) GetDependencies(object *unstructured.Unstructured) (dependencies []configv1alpha1.DependentObjectReference, enabled bool, err error) {
 	klog.V(4).Infof("Get dependencies of object: %v %s/%s with configurable interpreter.", object.GroupVersionKind(), object.GetNamespace(), object.GetName())
-	luaScript, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretDependency)
+
+	accessor, enabled := c.getCustomAccessor(object.GroupVersionKind())
 	if !enabled {
 		return
 	}
-	dependencies, err = c.luaVM.GetDependencies(object, luaScript)
+
+	scripts := accessor.GetDependencyInterpretationLuaScripts()
+	if scripts == nil {
+		enabled = false
+		return
+	}
+
+	referenceSet := newDependencySet()
```

Comment:
Seems we don't need to implement a set.
Let's check where to find a lib, or how to implement one with generic.